docs/source/examples/word-count.rst

```diff
@@ -0,0 +1,247 @@
+Word count in HDFS
+==================
+
+Setup
+-----
+
+In this example, we'll use ``distributed`` with the ``hdfs3`` library to count
+the number of words in text files (Enron email dataset, 6.4 GB) stored in HDFS.
+
+Copy data from S3 into HDFS:
+
+.. code-block:: bash
+
+   $ hadoop distcp s3n://{AWS_SECRET_ID}:{AWS_SECRET_KEY}@blaze-data/enron-email hdfs:///tmp/enron
+
+where ``AWS_SECRET_ID`` and ``AWS_SECRET_KEY`` are valid AWS credentials.
+
+Start the ``distributed`` scheduler and workers on the cluster.
+
+Code example
+------------
+
+Import ``distributed``, ``hdfs3``, and other standard libraries used in this example:
+
+.. code-block:: python
+
+   >>> import hdfs3
+   >>> from collections import defaultdict, Counter
+   >>> from distributed import Executor, progress
+
+Initalize a connection to HDFS, replacing ``NAMENODE_HOSTNAME`` and
+``NAMENODE_PORT`` with the hostname and port (default: 8020) of the HDFS
+namenode.
+
+.. code-block:: python
+
+   >>> hdfs = hdfs3.HDFileSystem('NAMENODE_HOSTNAME', port=NAMENODE_PORT)
```

Comment:
I prefer to be explicit here since we've only been testing within a certain type of setup, and the detection of config files may not be very robust for other setups.